include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
I'm new to Linux Tracepoints so apologies if i'm asking something that seems unreasonable, but i was thinking we could do something like this:

```
root@linux:~# cat /sys/kernel/debug/tracing/trace
# tracer: nop
#
# entries-in-buffer/entries-written: 1435/1435   #P:8
#
#                              _-----=> irqs-off
#                             / _----=> need-resched
#                            | / _---=> hardirq/softirq
#                            || / _--=> preempt-depth
#                            ||| /     delay
#           TASK-PID   CPU#  ||||    TIMESTAMP  FUNCTION
#              | |       |   ||||       |         |
 dbuf_evict-1422  [006] ....   956.984961: zfs_dbuf__evict__one: dbuf { spa "poolname" objset 0 object 63 level 0 blkid 0 offset 0 size 1024 state 4 holds 0 }
 dbuf_evict-1422  [006] ....   956.984980: zfs_dbuf__evict__one: dbuf { NULL }
 dbuf_evict-1422  [006] ....   956.984992: zfs_dbuf__evict__one: dbuf { spa "poolname" objset 0 object 57 level 0 blkid 0 offset 0 size 16384 state 4 holds 0 }
 dbuf_evict-1422  [006] ....   956.985003: zfs_dbuf__evict__one: dbuf { spa "poolname" objset 0 object 60 level 0 blkid 0 offset 0 size 16384 state 4 holds 0 }
 dbuf_evict-1422  [006] ....   956.985013: zfs_dbuf__evict__one: dbuf { NULL }
 dbuf_evict-1422  [006] ....   956.985028: zfs_dbuf__evict__one: dbuf { spa "poolname" objset 51 object 1 level 0 blkid 0 offset 0 size 512 state 4 holds 0 }
 dbuf_evict-1422  [006] ....   956.985037: zfs_dbuf__evict__one: dbuf { spa "poolname" objset 51 object 2 level 0 blkid 5 offset 655360 size 131072 state 4 holds 0 }
```

If i read the code correctly we are just skipping the Tracepoint when at least one parameter is NULL, which kind of defeat the purpose of keeping `DTRACE_PROBE2(dbuf__evict__one)` outside the `if (db != NULL)`.